lib/fluent/plugin/out_amplitude.rb

```diff
@@ -105,6 +107,17 @@ def extract_user_and_device!(amplitude_hash, record)
       end
     end
 
+    def extract_insert_id!(amplitude_hash, record)
```

Comment:
nit: this could be clearer using guard clauses:
```ruby
def extract_insert_id!(amplitude_hash, record)
  return unless @insert_id_key
  @insert_id_key.each do |insert_id_key|
    next unless record[insert_id_key]
    amplitude_hash[:insert_id] = record.delete(insert_id_key)
  end
end
```